testsuite/forge/src/backend/k8s/kube_api.rs

```diff
@@ -61,299 +61,148 @@ where
+    where
+        T: Clone + Metadata<Ty = ObjectMeta> + Send + Sync, // Ensure T has the necessary traits
+    {
+        pub fn new() -> Self {
+            MockK8sResourceApi {
+                resources: Arc::new(Mutex::new(BTreeMap::new())),
             }
-            Ok(self.config_map.clone())
         }
-    }
-
-    // Mock PersistentVolumeClaim API
 
-    pub struct MockPersistentVolumeClaimApi {
-        persistent_volume_claim: PersistentVolumeClaim,
-    }
-
-    impl MockPersistentVolumeClaimApi {
-        pub fn from_persistent_volume_claim(
-            persistent_volume_claim: PersistentVolumeClaim,
-        ) -> Self {
-            MockPersistentVolumeClaimApi {
-                persistent_volume_claim,
+        pub fn from_resource(resource: T) -> Self {
+            let resources = Arc::new(Mutex::new(BTreeMap::new()));
+            resources.lock().insert(
+                resource
+                    .metadata()
+                    .name
+                    .as_ref()
+                    .expect("Expected metadata to have name")
+                    .clone(),
+                resource.clone(),
+            );
+            MockK8sResourceApi { resources }
+        }
+
+        pub fn from_resources(resources_vec: Vec<T>) -> Self {
+            let resources = Arc::new(Mutex::new(BTreeMap::new()));
+            for resource in resources_vec {
+                resources.lock().insert(
+                    resource
+                        .metadata()
+                        .name
+                        .as_ref()
+                        .expect("Expected metadata to have name")
+                        .clone(),
+                    resource.clone(),
+                );
             }
+            MockK8sResourceApi { resources }
         }
     }
 
     #[async_trait]
-    impl ReadWrite<PersistentVolumeClaim> for MockPersistentVolumeClaimApi {
-        async fn get(&self, name: &str) -> Result<PersistentVolumeClaim, KubeError> {
-            if self.persistent_volume_claim.metadata.name == Some(name.to_string()) {
-                return Ok(self.persistent_volume_claim.clone());
-            }
-            return Err(KubeError::Api(ErrorResponse {
-                status: "failed".to_string(),
-                message: format!(
-                    "PersistentVolumeClaim with name {} could not be found in {:?}",
-                    name, self.persistent_volume_claim
-                ),
-                reason: "not_found".to_string(),
-                code: 404,
-            }));
-        }
-
-        async fn create(
-            &self,
-            _pp: &PostParams,
-            persistent_volume_claim: &PersistentVolumeClaim,
-        ) -> Result<PersistentVolumeClaim, KubeError> {
-            if self.persistent_volume_claim.metadata.name == persistent_volume_claim.metadata.name {
-                return Err(KubeError::Api(ErrorResponse {
+    impl<T> ReadWrite<T> for MockK8sResourceApi<T>
```

Comment:
I know I was the one who named this.... but can we call ReadWrite somethign more descriptive like "KubeApi" or something that indicates what it really is used for 🤦 my bad! old Perry was less wise